### PR TITLE
:lock: Authentication and User Accounts

### PR DESCRIPTION
**Pull request by awtkns**

### CODE REVIEW

File: src/server/auth.ts [hunk 1]
```diff
@@ -28,6 +31,19 @@ declare module "next-auth" {
   // }
 }
 
+const providers = [
+  GoogleProvider({
+    clientId: env.GOOGLE_CLIENT_ID,
+    clientSecret: env.GOOGLE_CLIENT_SECRET,
+    allowDangerousEmailAccountLinking: true,
```
Comment (asim-shrestha) [inline src/server/auth.ts:38]: Is this fine to leave as true? Will we be updating once we get a prod ready release

File: src/server/api/routers/agentRouter.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const agentRouter = createTRPCRouter({
+  create: protectedProcedure
+    .input(z.object({ name: z.string(), goal: z.string() }))
+    .mutation(async ({ input, ctx }) => {
+      const agent = await prisma.agent.create({
+        data: {
+          name: input.name,
+          goal: input.goal,
+          userId: ctx.session?.user?.id,
```
Comment (asim-shrestha) [inline src/server/api/routers/agentRouter.ts:14]: Probably change to NoSQL in the future?

**Review by asim-shrestha — APPROVED** [review]

File: src/server/auth.ts (hunk 1 above)
Comment (awtkns) [reply src/server/auth.ts:38]: This allow users to login with all providers and have them be be the same account. This is safe as long as the providers are well known and secure.

**Comment by hrada777** [comment]
Z

**Pull request merged**